### PR TITLE
[usage] added a test for ListUsage

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
adds a test for UsageService.ListUsage

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: components/usage/pkg/apiv1/usage_test.go [hunk 1]
```diff
@@ -259,3 +260,84 @@ func TestGetAndSetCostCenter(t *testing.T) {
...
+					Page:    1,
+				},
+			})
+			require.NoError(t, err)
+
+			require.Equal(t, test.creditsAtStart, metaData.CreditBalanceAtStart)
+			require.Equal(t, test.creditsAtEnd, metaData.CreditBalanceAtEnd)
+			require.Equal(t, test.recordsInRange, metaData.Pagination.Total)
+		})
+	}
```
Comment (easyCZ) [inline components/usage/pkg/apiv1/usage_test.go:341]:
```suggestion
	tests := []struct {
	    name string // test name
		start, end time.Time
		// expectations
		creditsAtStart float64
		creditsAtEnd   float64
		recordsInRange int64
	}{
		{start, end, 3, 10, 2},
		{end, end, 10, 10, 0},
		{start, start, 3, 3, 0},
		{start.Add(-200 * 24 * time.Hour), end, 0, 10, 4},
		{start.Add(-200 * 24 * time.Hour), end.Add(10 * 24 * time.Hour), 0, 20, 5},
	}

	for i, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			metaData, err := usageService.ListUsage(context.Background(), &v1.ListUsageRequest{
				AttributionId: string(attributionID),
				From:          timestamppb.New(test.start),
				To:            timestamppb.New(test.end),
				Order:         v1.ListUsageRequest_ORDERING_DESCENDING,
				Pagination: &v1.PaginatedRequest{
					PerPage: 1,
					Page:    1,
				},
			})
			require.NoError(t, err)

			require.Equal(t, test.creditsAtStart, metaData.CreditBalanceAtStart)
			require.Equal(t, test.creditsAtEnd, metaData.CreditBalanceAtEnd)
			require.Equal(t, test.recordsInRange, metaData.Pagination.Total)
		})
	}
```

It makes it nicer for humans on what you're testing, but also reflects it into the test when it fails

**Review by easyCZ — APPROVED** [review]
/hold in case you want to add name to the tests

File: components/usage/pkg/apiv1/usage_test.go (hunk 1 above)
Comment (svenefftinge) [reply components/usage/pkg/apiv1/usage_test.go:341]: It makes sense generally. In this case, I believe printing the number of the test is fine as well. But will definitely consider this in future!

**Pull request merged**